arkouda/io.py

```diff
@@ -309,404 +160,483 @@ def get_null_indices(filenames, datasets) -> Union[pdarray, Mapping[str, pdarray
+                "filenames": filenames,
+            },
+        )
+        rep = json.loads(rep_msg)  # See GenSymIO._buildReadAllMsgJson for json structure
+        _parse_errors(rep, allow_errors)
+        return _build_objects(rep)
 
-    >>> # Save using names instead of mapping
-    >>> ak.save_all([a, b], 'path/name_prefix', names=['a', 'b'], file_format='Parquet')
+
+def read_parquet(
+    filenames: Union[str, List[str]],
+    datasets: Optional[Union[str, List[str]]] = None,
+    iterative: bool = False,
+    strict_types: bool = True,
+    allow_errors: bool = False,
+) -> Union[
+    pdarray,
+    Strings,
+    arkouda.array_view.ArrayView,
+    Mapping[str, Union[pdarray, Strings, arkouda.array_view.ArrayView]],
+]:
     """
-    if names is not None:
-        if len(names) != len(columns):
-            raise ValueError("Number of names does not match number of columns")
-        else:
-            datasetNames = names
-    if isinstance(columns, dict):
-        pdarrays = list(columns.values())
-        if names is None:
-            datasetNames = list(columns.keys())
-    elif isinstance(columns, list):
-        pdarrays = cast(List[pdarray], columns)
-        if names is None:
-            datasetNames = [str(column) for column in range(len(columns))]
-    if mode.lower() not in ["append", "truncate"]:
-        raise ValueError("Allowed modes are 'truncate' and 'append'")
+    Read Arkouda objects from HDF5 file/s
+
+        Parameters
+        ----------
+        filenames : str, List[str]
+            Filename/s to read objects from
+        datasets : Optional str, List[str]
+            datasets to read from the provided files
+        iterative : bool
+            Iterative (True) or Single (False) function call(s) to server
+        strict_types: bool
+            If True (default), require all dtypes of a given dataset to have the
+            same precision and sign. If False, allow dtypes of different
+            precision and sign across different files. For example, if one
+            file contains a uint32 dataset and another contains an int64
+            dataset with the same name, the contents of both will be read
+            into an int64 pdarray.
+        allow_errors: bool
+            Default False, if True will allow files with read errors to be skipped
+            instead of failing.  A warning will be included in the return containing
+            the total number of files skipped due to failure and up to 10 filenames.
+
+        Returns
+        -------
+        For a single dataset returns an Arkouda pdarray, Arkouda Strings, or Arkouda ArrayView object
+        and for multiple datasets returns a dictionary of Arkouda pdarrays,
+        Arkouda Strings or Arkouda ArrayView.
+            Dictionary of {datasetName: pdarray or String}
+
+        Raises
+        ------
+        ValueError
+            Raised if all datasets are not present in all hdf5/parquet files or if one or
```

Comment:
Remove hdf5 from this error description since this is for parquet files only

Reply:
done